src/rules/__tests__/prefer-importing-jest-globals.test.ts

```diff
@@ -0,0 +1,451 @@
+      {
+        endColumn: 9,
+        column: 3,
+        line: 6,
+        messageId: 'preferImportingJestGlobal',
+      } as const,
+    ],
+  },
+  {
+    code: dedent`
+      console.log('hello');
+      const jestGlobals = require('@jest/globals');
+      describe("suite", () => {
+        test("foo");
+        expect(true).toBeDefined();
+      })
+      `,
+    output: dedent`
+      console.log('hello');
+      const { describe, test, expect } = require('@jest/globals');
+      describe("suite", () => {
+        test("foo");
+        expect(true).toBeDefined();
+      })
+    `,
+    errors: [
+      {
+        endColumn: 9,
+        column: 1,
+        line: 3,
+        messageId: 'preferImportingJestGlobal',
+      } as const,
+    ],
+  },
+  {
+    code: dedent`
+      const { pending } = require('actions');
+      describe('foo', () => {
+        test.each(['hello', 'world'])("%s", (a) => {});
+      });
+      `,
+    output: dedent`
+      const { pending } = require('actions');
+      const { describe, test } = require('@jest/globals');
+      describe('foo', () => {
+        test.each(['hello', 'world'])("%s", (a) => {});
+      });
+    `,
+    errors: [
+      {
+        endColumn: 9,
+        column: 1,
+        line: 2,
+        messageId: 'preferImportingJestGlobal',
+      } as const,
+    ],
+  },
+  {
+    code: dedent`
+      describe("suite", () => {
+        test("foo");
+        expect(true).toBeDefined();
+      })
+      `,
+    output: dedent`
+      const { describe, test, expect } = require('@jest/globals');
+      describe("suite", () => {
+        test("foo");
+        expect(true).toBeDefined();
+      })
+    `,
+    errors: [
+      {
+        endColumn: 9,
+        column: 1,
+        line: 1,
+        messageId: 'preferImportingJestGlobal',
+      } as const,
+    ],
+  },
+];
+
+ruleTester.run('prefer-importing-jest-globals', rule, {
+  valid: [
+    // {
+    //   code: dedent`
+    //     // with import
+    //     import { test, expect } from '@jest/globals';
+    //     test('should pass', () => {
+    //         expect(true).toBeDefined();
+    //     });
+    //     `,
+    //   parserOptions: { sourceType: 'module' },
+    // },
+    // {
+    //   code: dedent`
+    //     // with require
+    //     const { test, expect } = require('@jest/globals');
+    //     test('should pass', () => {
+    //         expect(true).toBeDefined();
+    //     });
+    //     `,
+    // },
+    // {
+    //   code: dedent`
+    //     import { it as itChecks } from '@jest/globals';
+    //     itChecks("foo");
+    //   `,
+    //   parserOptions: { sourceType: 'module' },
+    // },
+    // {
+    //   code: dedent`
+    //     const { test } = require('@jest/globals');
+    //     test("foo");
+    //   `,
+    // },
+    // {
+    //   code: dedent`
+    //     const { test } = require('my-test-library');
+    //     test("foo");
+    //   `,
+    // },
+  ],
+  invalid: [
+    ...importImports,
```

Comment:
I don't see any advantage to this and it gives you a worse typing & IDE experience since you'll get errors here rather than autocompletion when writing more cases (e.g. that's why you need to use `as const`) - can we just inline these?

Reply:
Removed, it was easier to deal with specific unit tests.